web/tests/user_groups.test.cjs

```diff
@@ -1148,3 +1150,45 @@ run_test("get_assigned_group_permission_object", () => {
         can_edit: true,
     });
 });
+
+run_test("check updation of group visibility status", () => {
+    const user_group_edit = zrequire("user_group_edit");
+    const params = {};
+    params.realm_user_groups = [
+        {...get_test_subgroup(1), deactivated: true},
+        {...get_test_subgroup(2), deactivated: false},
+        {...get_test_subgroup(3), deactivated: true},
+        {...get_test_subgroup(4), deactivated: true},
+        {...get_test_subgroup(5), deactivated: false},
+    ];
+    user_groups.init();
+    user_groups.initialize(params);
+    const ls = localstorage();
+
+    assert.ok(!$(".user-groups-list").hasClass("hide-deactived-user-groups"));
+    assert.ok(!$(".user-groups-list").hasClass("hide-active-user-groups"));
+
+    user_group_edit.update_displayed_groups(user_group_edit.FILTERS.ACTIVE_GROUPS);
+    assert.ok($(".user-groups-list").hasClass("hide-deactived-user-groups"));
+    assert.ok(!$(".user-groups-list").hasClass("hide-active-user-groups"));
+    assert.equal(
+        ls.get(user_group_edit.user_group_visibility_key),
+        user_group_edit.FILTERS.ACTIVE_GROUPS,
+    );
+
+    user_group_edit.update_displayed_groups(user_group_edit.FILTERS.DEACTIVATED_GROUPS);
+    assert.ok(!$(".user-groups-list").hasClass("hide-deactived-user-groups"));
+    assert.ok($(".user-groups-list").hasClass("hide-active-user-groups"));
+    assert.equal(
+        ls.get(user_group_edit.user_group_visibility_key),
+        user_group_edit.FILTERS.DEACTIVATED_GROUPS,
+    );
+
+    user_group_edit.update_displayed_groups(user_group_edit.FILTERS.ALL_GROUPS);
+    assert.ok(!$(".user-groups-list").hasClass("hide-deactived-user-groups"));
+    assert.ok(!$(".user-groups-list").hasClass("hide-active-user-groups"));
+    assert.equal(
+        ls.get(user_group_edit.user_group_visibility_key),
+        user_group_edit.FILTERS.ALL_GROUPS,
```

Comment:
I am not sure whether we should test this, since we usually do not have tests for such functions. And also, this module mainly contains data related tests.

Reply:
I meant the whole test, like we do not usually test these things like correct classes are added, removed, etc.

Reply:
I don't think there is actually any need to test this logic.